src/Filters/response.jl

```diff
@@ -58,6 +60,25 @@ function phasez(filter::FilterCoefficients, w = range(0, stop=π, length=250))
 end
 
 
+"""
+    grpdelay(fliter, w = range(0, stop=π, length=250))
+
+Group delay of a digital 'filter' at normalized frequency
+or frequencies 'w' in radians/sample.
+"""
+function grpdelay(filter::FilterCoefficients, w = range(0, stop=π, length=250))
+    filter = convert(PolynomialRatio, filter)
+    b, a = coefb(filter), filter.a.coeffs # reversed a
+    c = conv(b, conj(a))
+    cr = range(0, stop=length(c)-1) .* c
+    ejw = exp.(-im * w)
+    num = Polynomial(cr).(ejw)
+    den = Polynomial(c).(ejw)
+    gd = real(num ./ den) .- length(a) .+ 1
+    [x == Inf ? 0 : x for x in gd] # Set Infs to zeros
```

Comment:
Why is `Inf` replaced with `0`, but neither `-Inf` nor `NaN`? And why is it legitimate to replace it with zero anyway? IIUC, `gd` has non-finite values if the filter has poles and/or zeros on the unit circle. In both cases, there may well be a continuous extension of the group delay which will not necessarily be zero. E.g. for `filter = PolynomialRatio([1, 0, 1], [1])` and `w=π/2`, if it wasn't for numerical precision issues, one would get `num=-2` and `den=0`, hence either `gd=-Inf` (if `den==0.0`) or `gd=Inf` to be replaced with zero (if `den==-0.0`), while the only reasonable finite result would be `1`.

Reply:
I didn't think of those values. Looking at MATLAB it seems that they let Inf values occur in their implementation. SciPy checks to see if any values in ```den < 10 * EPSILON```. Should I just take this part out?

Reply:
Yes, I think returning `gd` should be fine. If the need arises, we can always figure out how to close any gaps.

Reply:
I added tests for type I-IV filters. I don't think that they pass. There are cases where NaN occurs from division by 0 + 0*im. Some Inf values also occur. I think I should check for values less than eps(Float64) and maybe make a case for linear phase FIR filters where it returns an array of the "by hand" values.